src/indicators/moving_standard_deviation.rs

```diff
@@ -0,0 +1,178 @@
+///
+/// # Exampile
+///
+/// ```
+/// use ta::indicators::MovingStandardDeviation;
+/// use ta::Next;
+///
+/// let mut msd = MovingStandardDeviation::new(3).unwrap();
+/// assert_eq!(msd.next(1.0), 1.4142135623730951);
+/// assert_eq!(msd.next(2.0), 1.1726039399558574);
+/// assert_eq!(msd.next(3.0), 0.816496580927726);
+/// ```
+///
+/// # Links
+///
+/// * [Standard Deviation, Wikipedia](https://en.wikipedia.org/wiki/Standard_deviation)
+///
+#[derive(Debug, Clone)]
+pub struct MovingStandardDeviation {
+    n: u32,
+    index: usize,
+    count: u32,
+    sum: f64,
+    vec: Vec<f64>,
+}
+
+impl MovingStandardDeviation {
+    pub fn new(n: u32) -> Result<Self> {
+        match n {
+            0 => Err(Error::from_kind(ErrorKind::InvalidParameter)),
+            _ => {
+                let indicator = Self {
+                    n: n,
+                    index: 0,
+                    count: 0,
+                    sum: 0.0,
+                    vec: vec![0.0; n as usize],
+                };
+                Ok(indicator)
+            },
+        }
+    }
+}
+
+impl Next<f64> for MovingStandardDeviation {
+    type Output = f64;
+
+    fn next(&mut self, input: f64) -> Self::Output {
+        self.index = (self.index + 1) % (self.n as usize);
+
+        let old_val = self.vec[self.index];
+        self.vec[self.index] = input;
+        if self.count < self.n {
+            self.count += 1;
+        }
+        self.sum = self.sum - old_val + input;
+        let mean = self.sum / (self.count as f64);
+        let mut mean_item_sum_pow = 0_f64;
+        for item in self.vec.iter() {
+            mean_item_sum_pow += (item - mean).powi(2);
+        }
+        (mean_item_sum_pow / (self.count as f64)).sqrt()
+    }
+}
+
+impl<'a, T: Close> Next<&'a T> for MovingStandardDeviation {
+    type Output = f64;
+
+    fn next(&mut self, input: &'a T) -> Self::Output {
+        self.next(input.close())
+    }
+}
+
+impl Reset for MovingStandardDeviation {
+    fn reset(&mut self) {
+        self.index = 0;
+        self.count = 0;
+        self.sum = 0.0;
+        for elem in self.vec.iter_mut() {
+            *elem = 0_f64;
+        }
+    }
+}
+
+impl Default for MovingStandardDeviation {
+    fn default() -> Self {
+        Self::new(9).unwrap()
+    }
+}
+
+impl fmt::Display for MovingStandardDeviation {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "SD({})", self.n)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::test_helper::*;
+
+    test_indicator!(MovingStandardDeviation);
+
+    #[test]
+    fn test_new() {
+        assert!(MovingStandardDeviation::new(0).is_err());
+        assert!(MovingStandardDeviation::new(1).is_ok());
+    }
+    
+    #[test]
+    fn test_next() {
+        let mut msd = MovingStandardDeviation::new(4).unwrap();
+        assert_eq!(msd.next(1.0), 1.7320508075688772);
+        assert_eq!(msd.next(2.0), 1.5811388300841898);
```

Comment:
According to [this online](http://www.alcula.com/calculators/statistics/standard-deviation/) tool SD of dataset `1, 2` must be `0.5`

![image](https://user-images.githubusercontent.com/113512/53672453-e1030f80-3c82-11e9-848b-5cddee06ac6b.png)


Looks like the implementation is not correct.

Reply:
This implementation was seeding population of n elements with `0_f64` instead of calculating only for inserted values. So this was going to properly function after n-th `next()` function call, I' am going to implement this correctly. Thanks for pointing this out.